src/Core/src/AbstractApi.php

```diff
@@ -151,11 +158,21 @@ protected function getSignerFactories(): array
      */
     private function getEndpoint(string $uri, array $query, ?string $region): string
     {
+        if (!$this->configuration->isDefault('endpoint')) {
+            $endpoint = $this->configuration->get('endpoint');
+        } else {
+            if (null === $region && !$this->configuration->isDefault('region')) {
+                $region = $this->configuration->get('region');
+            }
+            $endpoint = $this->getEndpointPattern($region) . $uri;
```

Comment:
The `$uri` will be added twice, both here and on 175.

Reply:
good catch thanks